coral-component-colorpicker/src/scripts/ColorArea.js

```diff
@@ -0,0 +1,560 @@
+   */   
+  get y() {
+    return this._y;
+  }
+  
+  set y(value) {
+    let rawY = Number(value, 10);
+    if(parseFloat(rawY).toFixed(3) !== parseFloat(this._y).toFixed(3)) {
+      if(isNaN(rawY)) {
+        rawY = this._minY();
+      }
+      this._y = this._snapValueToStep(rawY, this._minY(), this._maxY(),  this._stepY());
+      this._reflectAttribute('y', this._y);
+      this._elements.sliderY.setAttribute('value', this._y);
+      this.color = this._toHsvString(this._hue, this.x, this._y);
+    }
+  }
+  
+  /**
+   The ColorArea color string in hsla format.
+   @default hsla(0, 100%, 50%, 1)
+   @type {String}
+   @htmlattribute color
+   @htmlattributereflected
+   */   
+  get color() {
+    return colorUtil.toHslString(this._hue, new TinyColor({h:this._hue, s:this.x, v:this.y}).toHslString());
+  }
+  
+  set color(value) {
+    let color = new TinyColor(value);
+    if(!color.isValid) {
+      color = new TinyColor("hsla(120, 100%, 50%, 1)");
+      value = color.toHslString();
+    }
+    
+    // if  color strings are equal or colors are equivalent  
+    if(this.color === value || new TinyColor(this.color).toString(color.format) === color) {
+       return;
+    }
+
+    const {h,s,v} = colorUtil.extractHsv(value);
+    if(h !== this._hue) {
+      this._updateHue(colorUtil.getHue(value)); 
+    }
+    if(s !== this._x) {
+      this.x = s;
+    }
+    if(v !== this.Y) {
+      this.y = v;
+    }
+    this._updateHandle(this._hue, this.x, this.y, this.color);
+    this._reflectAttribute('color', this.color);
+  }
+        
+  /** @ignore */
+  static get observedAttributes() {
+    return super.observedAttributes.concat([
+      'label',
+      'x',
+      'y',
+      'disabled',
+      'color'
+    ]);
+  }  
+
+  /**
+   Whether this field is disabled or not.
+   @type {Boolean}
+   @default false
+   @htmlattribute disabled
+   @htmlattributereflected
+   */
+  get disabled() {
+    return this._disabled || false;
+  }
+
+  set disabled(value) {
+    this._disabled = transform.booleanAttr(value);
+    this._reflectAttribute('disabled', this._disabled);
+
+    this.classList.toggle('is-disabled', this._disabled);
+    this[this._disabled ? 'setAttribute' : 'removeAttribute']('aria-disabled', this._disabled);
+    this._elements.sliderX[this._disabled ? 'setAttribute' : 'removeAttribute']('disabled', this._disabled);
+    this._elements.sliderY[this._disabled ? 'setAttribute' : 'removeAttribute']('disabled', this._disabled);
+    this._elements.colorHandle[this._disabled ? 'setAttribute' : 'removeAttribute']('disabled', this._disabled);
+  }
+  
+  focus() {
+    this._sliderX.focus();
+  }
+  
+  /** @private */
+  _minX() {
+    return 0;
+  }
+  
+  /** @private */ 
+  _maxX() {
+    return 1;
+  }
+
+  _minY() {
+    return 0;
+  }
+  
+  /** @private */ 
+  _maxY() {
+    return 1;
+  }
+    
+  /** @private */ 
+  _stepX() {
+    return 0.01;
+  }
+
+  /** @private */ 
+  _stepY() {
+    return 0.01;
+  }
+  
```

Comment:
since all these are const values. you can add a getter like 

`get _minX() {
  return 0;
}`

and then you can use it as property rather than function

Reply:
The major difference I see b/w property and _minX function is:
Property can be changed by the consumer of the class while _minX function always returns the const value. 
We want to keep min, max, step values constant.

I don't know any benefit of accessing a property over a function call. Accessing a property is same as calling getter function.

Reply:
This corresponds more to a property rather than function. Generally the the property starting with underscore is private and should not be changed, in case consumer is changing its consumer faults. 

you can still avoid the change by  setting an empty setter

Reply:
To avoid getter/setter functions , we can simply define these constants in constructor.
e.g. 
constructor(){
 this._minX= 0;
 this._maxX = 100;
}

Properties are meant to be set/get by consumers to change the component's behaviour.